### PR TITLE
Cherry pick (#2517) to rel-1.29

### PR DESCRIPTION
**Pull request by ti-afra**
stabilise intrusion detection failures due to elastic search not being ready

* increase number of retries for intrusion-detection-es-job-installer

k8s < 1.26 does not support PodFailurePolicy API. To support more retries for older versions of kubernetes we need to set BackoffLimit to a higher number (default is 6) to prevent installation failures when responses from elastic search takes more time.

* add check for elasticsearch readiness to reconcile intrusion_detection

Check to make sure elasticsearch is in Ready phase as a pre-requisite to reconcile intrusion detection.

* update intrusiondetection_controller_test

* move the duplicated function GetElasticSearch to utils.go

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
